### PR TITLE
Add functional links to avatar dropdown

### PR DESCRIPTION
**Pull request by duranmla**
This PR adds functional links to the avatar dropdown whenever the user is logged in, besides, it also adds some extra test for `Session` component and integration test for the header component itself.

## Media
http://recordit.co/OOQCB0LUEg

## Aggregated work
After testing this has been realized that #3 were closed without having the logout functionality within the header so I've re-open the issue and I will make a PR to make sure to have that functionality as the previous version of the component.

### CODE REVIEW

File: src/__tests__/DebtcollectiveHeader.spec.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+      );
+
+      const userAvatar = await waitForElement(() =>
+        getByAltText(userData.username)
+      );
+
+      fireEvent.click(userAvatar);
+
+      expect(getByText(/profile/i).getAttribute("href")).toMatchInlineSnapshot(
+        "\"http://localhost:3000/u/johndoe\""
```
Comment (orlando) [inline src/__tests__/DebtcollectiveHeader.spec.js]: not sure if we should hardcode this, I think this will fail when running in Travis

**Comment by orlando** [comment]
just a small comment, we can leave it for now like that and refactor when CI is integrated

**Comment by duranmla** [comment]
We're mocking the value that environment variable is retrieving. So it should work, and if its not working then we make sure to mock the environment variable for that specific test

**Comment by orlando** [comment]
Cool, LGTM

**Pull request merged**